state.go

```diff
@@ -557,6 +560,98 @@ func (s *State) Channel(channelID string) (*Channel, error) {
 	return nil, ErrStateNotFound
 }
 
+// ThreadMemberUpdate updates the ThreadMember object for the user,
+// in a specific thread.
+func (s *State) ThreadMemberUpdate(t *ThreadMember) error {
+	if s == nil {
+		return ErrNilState
+	}
+
+	channel, ok := s.channelMap[t.ID]
+	if ok {
+		s.Lock()
+		defer s.Unlock()
+
+		channel.Member = t
+		return nil
+	}
+
+	return ErrStateNotFound
+}
+
+// ThreadMembersUpdate updates the member object of the thread.
+// If the user was removed,  the ThreadMember object in the state
+// will be set to nil.
+func (s *State) ThreadMembersUpdate(t *ThreadMembersUpdate) error {
+	if s == nil {
+		return ErrNilState
+	}
+
+	s.Lock()
+	defer s.Unlock()
+
+	thread, ok := s.channelMap[t.ID]
+	if ok {
+		for _, memberID := range t.RemovedMembersIDs {
+			if memberID == s.User.ID {
+				thread.Member = nil
+				break
+			}
+		}
+	}
+
```

Comment:
_I'm just lurking around since I'm waiting for this feature in the upstream and decided to give the PR a look._

Should not the method take into account `AddedMembers` field too? The documentation of the method says it `updates the member object of the thread` but it only takes care of the case when `the user was removed`.

Reply:
I think you are right! Will push an update. Let me know what you think.